lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -2534,16 +2543,28 @@ bool swift::diagnoseObjCMethodConflicts(SourceFile &sf) {
         if (auto accessor = dyn_cast<AccessorDecl>(originalMethod))
           originalDecl = accessor->getStorage();
 
+      // Conflicts between two imported ObjC methods are caused by the same
+      // clang decl having several Swift names.
+      if (originalDecl->hasClangNode() && conflictingDecl->hasClangNode())
+        continue;
+
+      bool breakingInSwift5 = false;
+      // Imported ObjC async methods weren't fully checked for selector
+      // conflicts until 5.7.
+      if (originalMethod->hasClangNode() && originalMethod->hasAsync())
```

Comment:
Does any special consideration need to be given to the ordering to make sure `originalMethod` ends up being the async alternative instead of the imported completion handler version here?

Reply:
You know, the more I look at how we select the “right” method, the less I trust it. I’m going to make this a little more robust.